### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by dorelo**
Bash code improvements

Possibly breaking changes? Who knows? Always test.

```
In amazon-bash-wifi.sh line 4:
Use ! grep -q instead of comparing output with [ -z .. ].


In amazon-bash-wifi.sh line 10:
read without -r will mangle backslashes.
b appears unused. Verify it or export it.


In amazon-bash-wifi.sh line 12:
Use $(..) instead of legacy `..`.


In amazon-bash-wifi.sh line 13:
$/${} is unnecessary on arithmetic variables.
```

### CODE REVIEW

**Comment by dorelo** [comment]
Updated with similar changes for amazon-bash.sh

File: amazon-bash-wifi.sh [hunk 1]
```diff
@@ -1,19 +1,19 @@
 #!/bin/bash
 MAC="<AMAZON DASH MAC ADDRESS>"
 
-if [ -z "$(ifconfig | grep 'mon0')" ]
+if "ifconfig | grep -q 'mon0'"
```
Comment (maxisme) [inline amazon-bash-wifi.sh:4]: Wrong. Shell checker messed this up!

**Review by maxisme — COMMENTED** [review]
Please alter

File: amazon-bash-wifi.sh (hunk 1 above)
Comment (maxisme) [reply amazon-bash-wifi.sh:4]: Need to leave this line as is. The rest works fine.

File: amazon-bash-wifi.sh [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 fi
 
 START=$(date +%s);
-tcpdump -l -i mon0 ether host "$MAC" | while read b; do
+tcpdump -l -i mon0 ether host "$MAC" | while read -r; do
         END=$(date +%s);
-        diff=`echo $((END-START)) | awk 'int($1%60)'` # Get time since last click.
-        if [ $(($diff)) -gt 2 ] # 2 seconds since last click as tcdump has multiple outputs.
+        diff=$(echo $((END-START)) | awk 'int($1%60)') # Get time since last click.
+        if [ "$diff" -gt 2 ] # 2 seconds since last click as tcdump has multiple outputs.
```
Comment (maxisme) [inline amazon-bash-wifi.sh:13]: removing `$(($diff))` causes errors

**Comment by maxisme** [comment]
Fixed myself

**Pull request closed without merging**